tests/flow/test_hnsw_tiered.py

```diff
@@ -0,0 +1,563 @@
+    hnsw_memory = bytes_to_mega(hnsw_index.index_memory())
+    print(f"total memory of hnsw index = {hnsw_memory} MB")
+    
+    # The index memory should be at least as the total memory of the vectors.
+    assert hnsw_memory > indices_ctx.get_vectors_memory_size()
+    
+    # Tiered index memory should be greater than HNSW index memory.
+    assert tiered_memory > hnsw_memory
+    execution_time_ratio = hnsw_index_time / tiered_index_time
+    print(f"with {threads_num} threads, insertion runtime is {round_(execution_time_ratio)} times better \n")
+    
+
+def search_insert(is_multi: bool, num_per_label = 1):
+    indices_ctx = IndexCtx(data_size=100000, is_multi=is_multi, num_per_label=num_per_label)
+    index = indices_ctx.tiered_index
+
+    num_labels = indices_ctx.num_labels
+    
+    query_data = indices_ctx.generate_queries(num_queries=1)
+    
+    # Add vectors to the flat index.
+    bf_index = indices_ctx.init_and_populate_flat_index()
+    
+    # Start background insertion to the tiered index.
+    index_start, _, _ = indices_ctx.populate_index(index)
+    
+    correct = 0
+    k = 10
+    searches_number = 0
+    print(f"hnsw size = {index.hnsw_label_count()}")
+    # run knn query every 1 s. 
+    total_tiered_search_time = 0
+    prev_bf_size = num_labels
+    while index.hnsw_label_count() < num_labels:
+        # For each run get the current hnsw size and the query time.
+        bf_curr_size = index.get_curr_bf_size()
+        query_start = time.time()
+        tiered_labels, _ = index.knn_query(query_data, k)
+        query_dur = time.time() - query_start
+        total_tiered_search_time += query_dur
+        
+        print(f"query time = {round_ms(query_dur)} ms")
+        
+        # BF size should decrease.
+        print(f"bf size = {bf_curr_size}")
+        assert bf_curr_size < prev_bf_size
+        
+        # Run the query also in the bf index to get the ground truth results.
+        bf_labels, _ = bf_index.knn_query(query_data, k)
+        correct += len(np.intersect1d(tiered_labels[0], bf_labels[0]))    
+        time.sleep(1)
+        searches_number += 1
+        prev_bf_size = bf_curr_size
+    
+    # HNSW labels count updates before the job is done, so we need to wait for the queue to be empty.
+    index.wait_for_index(1)
+    index_dur = time.time() - index_start
+    print(f"indexing during search in tiered took {round_(index_dur)} s")
+    
+    # Measure recall.
+    recall = float(correct)/(k*searches_number)
+    print("Average recall is:", round_(recall, 3))
+    print("tiered query per seconds: ", round_(searches_number/total_tiered_search_time)) 
+
+
+def test_create_tiered():
+    print("\nTest create tiered hnsw index")
+    create_tiered_index(is_multi=False)      
+      
+def test_create_multi():
+    print("Test create multi label tiered hnsw index")
+    create_tiered_index(is_multi=True, num_per_label=5)
+    
+def test_search_insert():
+    print(f"\nStart insert & search test")
```

Comment:
Add maybe the number of vectors and dim for this message

Reply:
Easier to print from the test itself